example.js

```diff
@@ -1,44 +1,58 @@
 const Ably = require("ably");
 
-const ApiKey = "INSERT-YOUR-API-KEY-HERE", /* Add your API key here */
-  clientId = "my-client-id"; /* This is who you will appear as in the presence set */
-if (ApiKey.indexOf("INSERT") === 0) { throw("Cannot run without an API key. Add your key to example.js"); }
+const apiKey = "YOUR_ABLY_API_KEY"; // Replace with your API key
+const randomId =
+  Math.random().toString(36).substring(2, 10) +
+  Math.random().toString(36).substring(2, 10);
+const realtime = new Ably.Realtime.Promise({
+  key: apiKey,
+  clientId: randomId, // Your ID in the presence set
+});
 
-console.log("Try running this script in multiple consoles concurrently, with a different clientId in each one. Each one will see the others in the presence set");
+async function doPresence() {
+  // Connect to Ably
+  await realtime.connection.once("connected");
+  console.log("Connected to Ably!");
 
-/* Instance the Ably library */
-var realtime = new Ably.Realtime({ key: ApiKey, clientId: clientId });
+  // Attach to the "chatroom" channel
+  const channel = realtime.channels.get("chatroom");
+  await channel.attach((err) => {
+    if (err) {
+      return console.error("Error attaching to the channel.");
+    }
+  });
 
-/* Enter the presence set of the 'chatroom' channel */
-var channel = realtime.channels.get('chatroom');
-channel.attach(function(err) {
-  if(err) { return console.log("Error attaching to the channel"); }
-  console.log('We are now attached to the channel');
-  channel.presence.enter('hello', function(err) {
-    if(err) { return console.error("Error entering presence"); }
-    console.log('We are now successfully present');
+  // Enter the presence set of the "chatroom" channel
+  await channel.presence.enter("hello", (err) => {
+    if (err) {
+      return console.error("Error entering presence set.");
+    }
+    console.log("This client has entered the presence set.");
   });
 
-  /* Every time the presence set changes, show the new set */
-  channel.presence.subscribe(function(presenceMsg) {
-    console.log('Received a ' + presenceMsg.action + ' from ' + presenceMsg.clientId);
-    channel.presence.get(function(err, members) {
-      if(err) { return console.log("Error fetching presence data: " + err); }
-      console.log("The presence set now consists of: " + members.map(function(member) {
-        return member.clientId;
-      }).join(", "));
+  // Subscribe to the presence set to receive updates
+  await channel.presence.subscribe((presenceMessage) => {
+    console.log(
+      "Presence update: " +
+        presenceMessage.action +
+        " from " +
+        presenceMessage.clientId
+    );
```

Comment:
Please use the argument list in the console.log instead of String concatenation.
```suggestion
    const { action, clientId } = presenceMessage;
    console.log( "Presence update: ", action, " from ", clientId );
```